addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -1247,6 +1244,49 @@ options.registry.AddField = FieldEditor.extend({
     },
 });
 
+options.registry.WebsiteFormSubmit = options.Class.extend({
+    /**
+     * Select the submit button's position
+     */
+    selectButtonPosition: function (previewMode, value, params) {
+        this.$target[0].querySelectorAll('.s_website_form_label').forEach(el => el.remove());
+        this.$target[0].classList.remove('text-center', 'text-left', 'text-right');
+        const button = this.$target[0].querySelector('.s_website_form_send');
+        switch (value) {
+            case 'align': {
+                const div = document.createElement('div');
+                const formOptions = this.el.closest('.o_we_customize_panel');
+                div.style.width = formOptions && formOptions.querySelector('we-customizeblock-options [data-select-style] input').value + 'px',
```

Comment:
This is in fact super ugly (I see this is also used in the original task work... why? Can't you read the form DOM instead of the editor one?)... Let's go with the standard select class solution. No JS, simply select class between text-left, text-right, text-center and s_website_form_label_aligned (something like that?). When the s_website_form_label_aligned is not put, the div is hidden in css.

Reply:
Leaving the div in will allow me to use the DOM to retrieve the label width. Before that there was the issue that all the fields could be removed leaving me with nowhere to retrieve the width from. I think that it is the right solution indeed !